### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/pantry/commit/a779fd09f6424067d58c96f0b337a575201759cc#commitcomment-81998596

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/pantry/commit/5d4c8dcadb945d2c1da76fabe7c70a7d20f22b60

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -50,7 +50,7 @@
 
 ## v0.5.1.1
 
-* Fix to allow mulitiple relative path of symlink.
```
Comment (jsoref) [inline ChangeLog.md:53]: Some projects don't like changing changelogs. I'm happy to drop this or any other change.

File: package.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name:        pantry
 version:     0.5.6
 synopsis:    Content addressable Haskell package management
-description: Please see the README on Github at <https://github.com/commercialhaskell/pantry#readme>
+description: Please see the README on GitHub at <https://github.com/commercialhaskell/pantry#readme>
```
Comment (jsoref) [inline package.yaml:4]: brand

File: src/Pantry/Storage.hs [hunk 3]
```diff
@@ -501,7 +501,7 @@ loadHackagePackageVersions
   -> ReaderT SqlBackend (RIO env) (Map P.Version (Map Revision BlobKey))
 loadHackagePackageVersions name = do
   nameid <- getPackageNameId name
-  -- would be better with esequeleto
+  -- would be better with esqueleto
```
Comment (jsoref) [inline src/Pantry/Storage.hs:504]: brand

File: src/Pantry/Types.hs [hunk 4]
```diff
@@ -38,7 +38,7 @@ module Pantry.Types
   , SafeFilePath
   , unSafeFilePath
   , mkSafeFilePath
-  , safeFilePathtoPath
+  , safeFilePathToPath
```
Comment (jsoref) [inline src/Pantry/Types.hs:41]: I'm assuming this is a local variable and that I can change it at will, this seems like a reasonable change, but I'm happy to drop.

File: src/Pantry/Types.hs [hunk 5]
```diff
@@ -1028,7 +1028,7 @@ instance Display PantryException where
     "Couldn't parse snapshot from " <> display sl <> ": " <> fromString e
   display (WrongCabalFileName pl sfp name) =
     "Wrong cabal file name for package " <> display pl <>
-    "\nCabal file is named " <> display sfp <>
+    "\nThe cabal file is named " <> display sfp <>
```
Comment (jsoref) [inline src/Pantry/Types.hs:1031]: This seems more in keeping w/ the style of this file.

File: src/Pantry/Types.hs [hunk 6]
```diff
@@ -1053,11 +1053,11 @@ instance Display PantryException where
     "\nActual:   " <> display mismatchActual
   display (UnknownArchiveType loc) = "Unable to determine archive type of: " <> display loc
   display (InvalidTarFileType loc fp x) =
-    "Unsupported tar filetype in archive " <> display loc <> " at file " <> fromString fp <> ": " <> displayShow x
+    "Unsupported tar file type in archive " <> display loc <> " at file " <> fromString fp <> ": " <> displayShow x
```
Comment (jsoref) [inline src/Pantry/Types.hs:1056]: The exception type thinks these are two words, and I agree.

File: src/Pantry/Types.hs [hunk 7]
```diff
@@ -1053,11 +1053,11 @@ instance Display PantryException where
...
   display (UnsupportedTarball loc e) =
     "Unsupported tarball from " <> display loc <> ": " <> display e
   display (NoHackageCryptographicHash ident) =
-    "Not cryptographic hash found for Hackage package " <> fromString (packageIdentifierString ident)
+    "No cryptographic hash found for Hackage package " <> fromString (packageIdentifierString ident)
```
Comment (jsoref) [inline src/Pantry/Types.hs:1060]: I actually tripped on this exception which is why I'm looking at this project... (the spell checker doesn't have an opinion on this at all)

File: src/Pantry/Types.hs [hunk 8]
```diff
@@ -1071,8 +1071,8 @@ instance Display PantryException where
   display (CRC32Mismatch loc fp Mismatch {..}) =
     "CRC32 mismatch in ZIP file from " <> display loc <>
     " on internal file " <> fromString fp <>
-    "\n.Expected: " <> display mismatchExpected <>
-    "\n.Actual:   " <> display mismatchActual
```
Comment (jsoref) [inline src/Pantry/Types.hs:1075]: This didn't fit the style elsewhere...

File: src/Pantry/Types.hs [hunk 9]
```diff
@@ -1941,10 +1941,10 @@ parseRawSnapshotLocation :: Text -> Unresolved RawSnapshotLocation
 parseRawSnapshotLocation t0 = fromMaybe (parseRawSnapshotLocationPath t0) $
   (either (const Nothing) (Just . pure . RSLCompiler) (parseWantedCompiler t0)) <|>
   (pure <$> RSLSynonym <$> parseSnapName t0) <|>
-  parseGithub <|>
+  parseGitHub <|>
   parseUrl
   where
-    parseGithub = do
+    parseGitHub = do
```
Comment (jsoref) [inline src/Pantry/Types.hs:1947]: Again, I'm hoping it's a local, but happy to drop if it's a problem.

**Review by snoyberg — APPROVED** [review]
Thanks!

**Pull request merged**